x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichStore.java

```diff
@@ -86,6 +86,7 @@ public static void putPolicy(
                 String[] concreteIndices = indexNameExpressionResolver.concreteIndexNames(
                     current,
                     IndicesOptions.strictExpandOpen(),
+                    true,
```

Comment:
This is not a blocker for this PR, but I'm curious to hear what your take is on Martijn's [comment](https://github.com/elastic/elasticsearch/issues/98836#issuecomment-1692650512) about possible performance issues when it comes to using aliases and data streams in enrich. I would think this is a pretty realistic issue, but maybe I'm missing something. I don't see any tip or warning [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/enrich-setup.html) or [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest-enriching-data.html), maybe it's worth adding one over there?

Reply:
I do think it's worth documenting that enrich indices are not expected to be large. I'll do that in a follow-up PR. As for Martijn's comment, yes, unfortunately large amounts of data probably isn't the best idea for enrich indices, but we already have that issue with allowing wildcard patterns (someone could use `logs-*` and grab hundreds of indices already), so I don't think we're making anything worse here.